### PR TITLE
added nextjs 13 app router native support & docs improvements.

### PR DESCRIPTION
**Pull request by KevinEdry**
## Summary
This PR aims to revise and add two main things:
- Bring native support for Nextjs 13 app router to Quirrel - https://github.com/quirrel-dev/quirrel/issues/1138
- Improve the docs experience since the hosted version of Quirrel isn't available anymore

## Nextjs 13 App Router
Because Nextjs 13 now expects a named method function as a route handler (POST, GET etc..), we need to provide it as part of the `Queue` and `CronJob` methods, we also want the queue instances themselves, so I've made it so it would be accessible via object destructuring, that way we can export both `POST` and the queue instance in one go, and alias the queue instance with  a meaningful name.

Queue Usage:
```typescript
  export const { POST, queue: sampleQueue } = Queue(
   "/api/queues/sample",
   (payload) => {
       ...logic goes here
  });
 
 await sampleQueue.enqueue({ jobData });
```

CronJob Usage:
```typescript
  export const { POST } = CronJob(
   "/api/queues/sample",
   "0 0 * * 1"
   (payload) => {
       ...logic goes here
  });
```
I've added some docs changes as well to explain the usage of the new queue api.

**The object destructuring is the best way I could think of to keep the named export function `POST` and still have a meaningful name for the queue instance, if you can think of a better implementation for this please let me know!**

### code
- added new `next13.ts` file to support nextjs 13 app directory natively.

### docs
- added to proper documentation for all of the Nextjs changes.
- added new "Development" category which explains how to run the Quirrel instance in a docker compose, or as a standalone instance.
- added  a new  and improved "Deployment" category with several guides on how to deploy quirrel (docker, railway, fly and vercel).
- removed the old "Deploy" page.

### CODE REVIEW

**Comment by Skn0tt** [comment]
Hi @KevinEdry, this is an amazing PR! Thank you so much - I love how you improved the docs. It's quite big, so reviewing will take a couple iterations, but we'll get there.

File: src/next-app.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ *   "/api/cronjobs/sample",
+ *   "0 0 * * 1"
+ *   (payload) => {
+ *      ...logic goes here
+ *   }
+ * );
+ * 
+ * export const POST = sampleCron;
+ */
+export function CronJob(
```
[the remark this answers is not in the document]
Comment (Skn0tt) [reply src/next-app.ts:94]: CronJobs don't receive any kind of payload, so there's no need for this.

**Comment by Skn0tt** [comment]
in https://github.com/quirrel-dev/quirrel/pull/1141/commits/1d51c94ab121f8e24505a3aaaa1cd7882df0919f, i've removed the docker-compose development docs. Quirrel has a local dev mode, and that should be used locally. I think that docker-compose is way too complex for most usecases of this, so I don't want to highlight it in the docs. Appreciate the document, though!

**Comment by KevinEdry** [comment]
@Skn0tt I wonder if we also need to change the cron-detector since we are no longer importing from `quirrel/next`.

**Comment by Skn0tt** [comment]
> but I do think the docs should also reflect the implementation for every framework

I want to keep the documentation small and concise, and i'm unsure if another documentation for `CronJob` for every single framework adds much value.

**Comment by Skn0tt** [comment]
> I wonder if we also need to change the cron-detector

I don't think so, the cron-detector isn't hard-coded on framework names, but matches on `quirrel/*`: https://github.com/quirrel-dev/quirrel/blob/305e4243a9286f1aed3c6792063790d07d946a9d/src/cli/cron-detector.ts#L20

**Comment by Skn0tt** [comment]
This is now in a state where i'm happy with it. @KevinEdry could you give it another read and let me know if there's anything you think needs changing?

File: src/next-app.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (KevinEdry) [reply src/next-app.ts:94]: I use a cronjob with a payload, the cron is starting the job, the job is making an api request, and if the response contains a cursor, it invokes the same job with the cursor:

```typescript
export const { POST, queue: companiesQueue } = CronJob<Payload>(
  "/api/queue/companies",
  "0 0 * * 1",
  async (payload?: Payload) => {
    try {
      const fetchUrl =
        payload?.cursor != null
          ? payload.cursor
          : "https://api.random-example.com/something";

      const res = await fetchApi<Ticker>(fetchUrl);
      const { results, next_url } = tickersSchema.parse(res);

      if (next_url != null) {
        companiesQueue.enqueue({ cursor: next_url });
      }

      return;
    } catch (error) {
      console.log({ error });
      throw error;
    }
  }
);
```

**Comment by KevinEdry** [comment]
@Skn0tt LGTM, the only thing I would change is to add the `Payload` to the CronJob, but it is your call, I will manage either way in my project and my specific use-case.

File: src/next-app.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Skn0tt) [reply src/next-app.ts:94]: Interesting! I'd have never imagined anyone to use a CronJob like this 😁 I think if we documented this, it'd be very non-obvious how it works in most cases. So i'd prefer to not document it. I'm sure you'll find a workaround for your specific case :D

**Review by Skn0tt — APPROVED** [review]

**Pull request merged**